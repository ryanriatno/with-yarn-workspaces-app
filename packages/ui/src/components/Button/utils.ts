export const getColorButton = (color: string): string => {
  let cssClass: string = ''

  switch (color) {
    case 'blue':
      cssClass =
        'bg-clrPrimary hover:bg-clrPrimary01 text-white font-bold py-2 px-4 rounded w-full transition-all duration-300 ease-in'
      break
    case 'white':
      cssClass =
        'bg-clrInverse hover:bg-clrPrimary text-clrPrimary hover:text-white font-bold py-2 px-4 rounded w-full transition-all duration-300 ease-in border-solid border border-clrPrimary'
      break
    case 'social-white':
      cssClass =
        'bg-clrInverse hover:bg-clrField01 text-sm font-thin text-clrSubText5 py-2 px-4 rounded w-full transition-all duration-300 ease-in border-solid border border-clrBorder01 flex items-center justify-center'
      break
    case 'social-black':
      cssClass =
        'bg-black hover:bg-gray-700 text-sm font-thin text-clrInverse py-2 px-4 rounded w-full transition-all duration-300 ease-in border-solid border border-clrBorder01 flex items-center justify-center'
      break
  }

  return cssClass
}
