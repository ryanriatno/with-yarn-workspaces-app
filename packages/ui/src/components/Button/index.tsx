import React from 'react'
import { getColorButton } from './utils'

interface Props {
  style?: Record<string, unknown>
  loadingButton?: boolean
  color: string
  onClick?: () => void
  prefix?: any
  children: React.ReactNode
}

export const Button: React.FC<Props> = (props) => {
  return (
    <button
      onClick={props.onClick}
      className={getColorButton(props.color)}
      style={props.style}
    >
      {props.prefix}
      {props.loadingButton ? (
        <svg className="inline-block w-4 h-4 fill-current">
          <svg version="1.1" id="L9" x="0px" y="0px" viewBox="25 25 50 50">
            <path d="M73,50c0-12.7-10.3-23-23-23S27,37.3,27,50 M30.9,50c0-10.5,8.5-19.1,19.1-19.1S69.1,39.5,69.1,50">
              <animateTransform
                attributeName="transform"
                attributeType="XML"
                type="rotate"
                dur="1s"
                from="0 50 50"
                to="360 50 50"
                repeatCount="indefinite"
              />
            </path>
          </svg>
        </svg>
      ) : (
        props.children
      )}
    </button>
  )
}

export default Button
