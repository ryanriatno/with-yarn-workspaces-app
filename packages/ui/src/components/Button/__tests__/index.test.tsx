import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, getByText, fireEvent } from '@testing-library/react'
import Button from '../index'

describe('<Button /> test case', () => {
  describe('Blue button test case', () => {
    const buttonText = 'Blue button'
    const { container } = render(<Button color="black">{buttonText}</Button>)

    const linkElement = getByText(container, buttonText)

    test(`Render ${buttonText} correctly`, () => {
      expect(container).toMatchSnapshot()
    })

    test(`Have a text ${buttonText}`, () => {
      expect(linkElement).toBeInTheDocument
    })

    test(`${buttonText} clicked`, () => {
      const clicked = fireEvent.click(linkElement)
      expect(clicked).toBe(true)
    })
  })

  describe('White button test case', () => {
    const buttonText = 'White button'
    const { container } = render(<Button color="black">{buttonText}</Button>)

    const linkElement = getByText(container, buttonText)

    test(`Render ${buttonText} correctly`, () => {
      expect(container).toMatchSnapshot()
    })

    test(`Have a text ${buttonText}`, () => {
      expect(linkElement).toBeInTheDocument
    })

    test(`${buttonText} clicked`, () => {
      const clicked = fireEvent.click(linkElement)
      expect(clicked).toBe(true)
    })
  })

  describe('Black button test case', () => {
    const buttonText = 'Black button'
    const { container } = render(<Button color="black">{buttonText}</Button>)

    const linkElement = getByText(container, buttonText)

    test(`Render ${buttonText} correctly`, () => {
      expect(container).toMatchSnapshot()
    })

    test(`Have a text ${buttonText}`, () => {
      expect(linkElement).toBeInTheDocument
    })

    test(`${buttonText} clicked`, () => {
      const clicked = fireEvent.click(linkElement)
      expect(clicked).toBe(true)
    })
  })
})
