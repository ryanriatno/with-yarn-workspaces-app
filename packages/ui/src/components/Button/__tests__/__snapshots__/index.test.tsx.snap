// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> test case Black button test case Render Black button correctly 1`] = `<div />`;

exports[`<Button /> test case Blue button test case Render Blue button correctly 1`] = `
<div>
  <button>
    Blue button
  </button>
</div>
`;

exports[`<Button /> test case White button test case Render White button correctly 1`] = `<div />`;
